Update KSM documentation regarding nodes.by_condition (#17823)
DataDog/documentation: content/en/integrations/kubernetes_state_core.md

The Kubernetes State Metrics Core check is not backward compatible, be sure to read the changes carefully before migrating from the legacy `kubernetes_state` check.

`kubernetes_state.node.by_condition`

> : A new metric with node name granularity. The legacy metric `kubernetes_state.nodes.by_condition` has been depricated in favor of this one. **Note:** This metric was backported into the Legacy check, where both it and the metric it replaces (`nodes.by_condition`) are available. 

Reviewer: ```suggestion
: A new metric with node name granularity. The legacy metric `kubernetes_state.nodes.by_condition` is deprecated in favor of this one. **Note:** This metric is backported into the Legacy check, where both metrics (it and the legacy metric it replaces) are available. 
```
